transport_nantes/topicblog/views.py

```diff
@@ -1045,15 +1067,30 @@ def beacon_view(response, **kwargs):
 
     # We remove the ".gif" from the end of the token
     token = kwargs['token'][:-4]
-    email, send_record_id = token_valid(token)
-    if email and send_record_id:
-        # In case the token is valid, we update the open_time
-        send_record: SendRecordMarketingEmail
-        send_record = SendRecordMarketingEmail.objects.get(pk=send_record_id)
-        if not send_record.open_time:
-            send_record.open_time = datetime.now(timezone.utc)
-            send_record.save()
-            logger.info(f"{email} opened email (SR.id :{send_record.pk})")
+    send_record_class_string, send_record_id = token_valid(token)
+    if send_record_class_string:
+        send_record_class = \
+            apps.get_model("topicblog", send_record_class_string)
+        try:
+            if not send_record_id:
+                raise ValueError
+            # In case the token is valid, we update the open_time
+            send_record = send_record_class.objects.get(pk=send_record_id)
+        except ObjectDoesNotExist:
+            logger.info(f"No send record with id {send_record_id} in class "
+                        f"{send_record_class_string} has been found")
+        except ValueError:
+            logger.info(
```

Comment:
You are catching the valueError you just threw?  Why don't you simply handle it where you threw it and exit right away?  It saves state for the reader.